language/compiler/ir_to_bytecode/src/compiler.rs

```diff
@@ -283,6 +288,23 @@ impl<'a> CompilationScope<'a> {
             if self.modules[idx].name() == module_name && self.modules[idx].address() == addr {
                 self.imported_modules
                     .insert(import_name.to_string(), (idx as u8, mh_idx));
+
+                self.struct_definitions.insert(
+                    import_name.to_string(),
+                    self.modules[idx]
+                        .struct_defs()
+                        .iter()
+                        .enumerate()
+                        .map(|(sd_idx, def)| {
+                            let sh = self.modules[idx].struct_handle_at(def.struct_handle);
+                            let name = self.modules[idx].string_at(sh.name);
+                            (
+                                name.to_string(),
```

Comment:
Nit: `StructDefinitionView::new(self.modules[idx], def).name().to_string()` would be a bit shorter

Reply:
Views aren't used anywhere in the compiler so I'd be reluctant to introduce such a dependency just for this simple case. But in general I agree that many of the internal APIs of the compiler can and should be improved.